lib/ansible/modules/identity/ipa/ipa_subca.py

```diff
@@ -0,0 +1,246 @@
+
+    def subca_del(self, subca_name=None):
+        return self._post_json(method='ca_del', name=subca_name)
+
+    def subca_disable(self, subca_name=None):
+        return self._post_json(method='ca_disable', name=subca_name)
+
+    def subca_enable(self, subca_name=None):
+        return self._post_json(method='ca_enable', name=subca_name)
+
+
+def get_subca_dict(details=None):
+    module_subca = dict()
+    if details['description'] is not None:
+        module_subca['description'] = details['description']
+    if details['subca_subject'] is not None:
+        module_subca['ipacasubjectdn'] = details['subca_subject']
+    return module_subca
+
+
+def get_subca_diff(client, ipa_subca, module_subca):
+    details = get_subca_dict(module_subca)
+    return client.get_diff(ipa_data=ipa_subca, module_data=details)
+
+
+def ensure(module, client):
+    subca_name = module.params['subca_name']
+    subca_subject_dn = module.params['subca_subject']
+    subca_desc = module.params['subca_desc']
+
+    state = module.params['state']
+
+    ipa_subca = client.subca_find(subca_name)
+    module_subca = dict(description=subca_desc,
+                        subca_subject=subca_subject_dn)
+
+    changed = False
+    if state == 'present':
+        if not ipa_subca:
+            changed = True
+            if not module.check_mode:
+                client.subca_add(subca_name=subca_name, subject_dn=subca_subject_dn, details=module_subca)
+        else:
+            diff = get_subca_diff(client, ipa_subca, module_subca)
+            # IPA does not allow to modify Sub CA's subject DN
+            # So skip it for now.
+            if 'ipacasubjectdn' in diff:
+                diff.remove('ipacasubjectdn')
+                del module_subca['subca_subject']
+
+            if len(diff) > 0:
+                changed = True
+                if not module.check_mode:
+                    client.subca_mod(subca_name=subca_name, diff=diff, details=module_subca)
+    elif state == 'absent':
+        if ipa_subca:
+            changed = True
+            if not module.check_mode:
+                client.subca_del(subca_name=subca_name)
+    elif state == 'disable':
+        if ipa_subca:
+            changed = True
+            if not module.check_mode:
+                client.subca_disable(subca_name=subca_name)
+    elif state == 'enable':
+        if ipa_subca:
+            changed = True
+            if not module.check_mode:
+                client.subca_enable(subca_name=subca_name)
+
+    return changed, client.subca_find(subca_name)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            subca_name=dict(type='str', required=True, aliases=['name']),
+            subca_subject=dict(type='str', required=True),
+            subca_desc=dict(type='str'),
+            state=dict(type='str', required=False, default='present', choices=['present', 'absent', 'enable', 'disable']),
```

Comment:
Since these refer to state, I think it would make more sense to have `enabled` and `disabled` (instead of `enable`/`disable`)

Reply:
ok